remote-trait-object-tests/src/mod_ping/requester.rs

```diff
@@ -0,0 +1,38 @@
+// Copyright 2020 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pub use super::traits::PingInterface;
+use remote_trait_object::Port;
+
+pub struct PingRequester<'a> {
+    port: &'a Port,
```

Comment:
It won't be possible to hold the port by reference soon, if you find more complicated pattern (e.g Context contains instance of this with a `Arc<dyn>`). Note that it is `std::sync::Weak<dyn Port>` in my `no-global` branch.
You can change this later.

Reply:
Yes, I'll change this code soon. I don't want to consider the lifetime of Port yet. I want to delay the lifetime problem.
Thanks for giving the referenced for me. I'll read the no-global branch later.